apps/wallpaper/js/share.js

```diff
@@ -26,9 +26,11 @@ window.onload = function() {
   var scale;
   var previewImage = document.getElementById('previewImage');
 
-  // How big (in device pixels) is the screen?
-  var screenWidth = Math.ceil(screen.width * window.devicePixelRatio);
-  var screenHeight = Math.ceil(screen.height * window.devicePixelRatio);
+   // How big (in device pixels) is the screen?
+  var screenWidth = Math.ceil(Math.min(screen.width, screen.height) *
+                    window.devicePixelRatio);
+  var screenHeight = Math.ceil(Math.max(screen.width, screen.height) *
+                     window.devicePixelRatio);
 
```

Comment:
This fixes intermittent IndexSizeError  seen at Line141 due to incorrect screen.width and screen.height.

01-02 07:14:45.360 W/Wallpaper(10156): [JavaScript Error: "IndexSizeError: Index or size is negative or greater than the allowed amount" {file: "app://wallpaper.gaiamobile.org/js/share.js" line: 141}]